### PR TITLE
Enhance manim banner animations

### PR DESCRIPTION
**Pull request by AntonBallmaier**
## Changelog / Overview
The ManimBanner now has cleaner default animations. 

## Motivation
Manim is an animation Library / Framework. I think the logo animations should be as good as possible :slightly_smiling_face:  

## Explanation for Changes
The new animations look better (imo). The `create` and `expand` methods can take a `run_time` argument. Also the `expand` method takes a new optional parameter controlling the direction in which the logo is expanded.

A more detailed list of the visual changes:
**Create animation:**
- The animation takes 2s (by default) instead of 2.1s (fixed)
- The shapes start off with velocity instead of accelerating from 0.
- The shapes fade in quickly instead of just jumping into existence.
- Some timing changes

**Expand animation:**
- The letters "anim" get uncovered by the shifting shapes
- The shapes overshoot the final position slightly and shift back behind the letters.
- Some timing changes 

https://user-images.githubusercontent.com/26899845/114272828-faf1ad80-9a17-11eb-8d1f-3aaf4ba8fbe2.mp4

https://user-images.githubusercontent.com/26899845/114272830-fc22da80-9a17-11eb-9941-00e4b36acf9b.mp4

## Checklist
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [x] I have written a descriptive PR title (see top of PR template for examples)
- [x] I have written a changelog entry for the PR or deem it unnecessary
- [x] My new functions/classes either have a docstring or are private
- [ ] My new functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [x] My new documentation builds, looks correctly formatted, and adds no additional build warnings

## Reviewer Checklist
- [ ] The PR title is descriptive enough
- [ ] The PR is labeled correctly
- [ ] The changelog entry is completed if necessary
- [ ] Newly added functions/classes either have a docstring or are private
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings

### CODE REVIEW

**Comment by behackl** [comment]
I don't think this is related to your PR. I'm investigating.

**Comment by behackl** [comment]
(The animations are *amazing*, by the way. I'd like to get this merged as soon as possible, thank you for your efforts!)

**Comment by behackl** [comment]
> I don't think this is related to your PR. I'm investigating.

It unfortunately *is* related to your PR: I've added a debug print to `slide_and_uncover`, this is what it returns:

```
called for ManimBanner of 3 submobjects and 0
Animation 1: Succession(Group):   0%|                                                                                                                                                                                 | 0/45 [00:00<?, ?it/s]called for ManimBanner of 7 submobjects and 0.0
called for ManimBanner of 13 submobjects and 0.00014814814814814815
called for ManimBanner of 25 submobjects and 0.0011851851851851852
called for ManimBanner of 49 submobjects and 0.004000000000000001
called for ManimBanner of 97 submobjects and 0.009481481481481481
called for ManimBanner of 193 submobjects and 0.018518518518518517
called for ManimBanner of 385 submobjects and 0.03200000000000001
called for ManimBanner of 769 submobjects and 0.05081481481481482
called for ManimBanner of 1537 submobjects and 0.07585185185185185
Animation 1: Succession(Group):  20%|█████████████████████████████████▊                                                                                                                                       | 9/45 [00:00<00:00, 75.85it/s]called for ManimBanner of 3073 submobjects and 0.108
called for ManimBanner of 6145 submobjects and 0.14814814814814814
called for ManimBanner of 12289 submobjects and 0.19718518518518513
called for ManimBanner of 49155 submobjects and 0.25600000000000006
called for ManimBanner of 196617 submobjects and 0.32548148148148154
called for ManimBanner of 786465 submobjects and 0.40651851851851856
```

The problematic part of your implementation is

```py
                if mob.square.get_center()[0] > letter.get_center()[0]:
                    letter.set_opacity(1)
                    self.add_to_back(letter)
```

where letter seems to be added repeatedly.

**Comment by AntonBallmaier** [comment]
Is is in turn based on this bug (#1099)... I thought it was fixed by now :laughing:
Ok, easy fix...

File: manim/mobject/logo.py [hunk 1]
```diff
@@ -71,13 +67,15 @@ def __init__(self, dark_theme: bool = True):
         self.circle = Circle(color=logo_green, fill_opacity=1).shift(LEFT)
         self.square = Square(color=logo_blue, fill_opacity=1).shift(UP)
         self.triangle = Triangle(color=logo_red, fill_opacity=1).shift(RIGHT)
-        self.add(self.triangle, self.square, self.circle, self.M)
+        self.shape = VGroup(self.triangle, self.square, self.circle)
```
Comment (behackl) [inline manim/mobject/logo.py:70]: Just a question of semantics: should `self.shape` be called `self.shapes`?

File: manim/mobject/logo.py [hunk 2]
```diff
@@ -71,13 +67,15 @@ def __init__(self, dark_theme: bool = True):
...
+        self.add(self.shape, self.M)
         self.move_to(ORIGIN)
 
         anim = VGroup()
         for i, ch in enumerate("anim"):
             tex = Tex(
                 "\\textbf{" + ch + "}",
+                # ch,
```
Comment (behackl) [inline manim/mobject/logo.py:78]:
```suggestion
```

File: manim/mobject/logo.py [hunk 3]
```diff
@@ -174,34 +165,97 @@ def expand(self) -> Succession:
...
+        --------
+        .. manim:: ExpandDirections
+
+            class ExpandDirections(Scene):
+                def construct(self):
+                    banners = [ManimBanner().scale(0.5).shift(UP*x) for x in [-2, 0, 2]]
+                    self.play(
+                        banners[0].expand(direction = "right"),
+                        banners[1].expand(direction = "center"),
+                        banners[2].expand(direction = "left"),
```
Comment (behackl) [inline manim/mobject/logo.py:190]:
```suggestion
                        banners[0].expand(direction="right"),
                        banners[1].expand(direction="center"),
                        banners[2].expand(direction="left"),
```

**Review by behackl — APPROVED** [review]
Thank you very much, I was a bit sad to see our Banner animation broken when I looked at it again recently -- but with this PR, the animations are constructed way cleaner and I generally like the way the banner is animated much more.

Very well done, good job! I only have some minor semantic and syntactic remarks.

File: manim/mobject/logo.py (hunk 1 above)
Comment (AntonBallmaier) [reply manim/mobject/logo.py:70]: Yeah, thought about that as well. `shapes` would be nicer imo.

Comment (behackl) [reply manim/mobject/logo.py:70]: I'd also be in favor of refactoring. Could you change the identifier name, or should I add a commit?

**Review by naveen521kk — APPROVED** [review]

**Pull request merged**